src/m365/spo/commands/site/site-recyclebinitem-list.ts

```diff
@@ -0,0 +1,91 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  type?: number;
+  secondary?: boolean;
+}
+
+class SpoSiteRecycleBinItemListCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_RECYCLEBINITEM_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists items from recycle bin';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['Id', 'Title', 'DirName'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving all items from recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    const state: string = args.options.secondary ? '2' : '1';
+
+    let requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin?$filter=(ItemState eq ${state})`;
+
+    if (typeof args.options.type !== 'undefined') {
+      requestUrl += ` and (ItemType eq ${args.options.type})`;
+    }
+
+    const requestOptions: any = {
+      url: requestUrl,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    request
+      .get<any>(requestOptions)
+      .then((recycleBinItemProperties: any): void => {
+        logger.log(recycleBinItemProperties.value);
```

Comment:
As we're referring to `.value` here, let's define the request's return type as `{ value: any[] }` to avoid refactoring issues in the future. In my original remark it got lost on me that the response is an array wrapped in `value`, sorry for that.

Reply:
a ok. Now I think I understand. will do

Reply:
ok corrected. I hope I done it right this time 🙂